components/filter/filter-dimension-set.js

```diff
@@ -173,7 +167,16 @@ class FilterDimensionSet extends LitElement {
 	_getSlottedNodes() {
 		if (!this._slot) return [];
 		const nodes = this._slot.assignedNodes({ flatten: true });
-		return nodes.filter((node) => node.nodeType === Node.ELEMENT_NODE && node.tagName.toLowerCase() === 'd2l-filter-dimension-set-value');
+		return nodes.filter((node) => {
+			if (node.nodeType !== Node.ELEMENT_NODE) return false;
+
+			const hasDateFilter = node.tagName.toLowerCase() === 'd2l-filter-dimension-set-date-text-value';
```

Comment:
We have these types tightly under wraps - it's not like anyone can just create their own value types. Nevertheless, it feels a little off to have `d2l-filter-dimension-set` and `d2l-filter` be aware of the different value types. To avoid that, at least for this bit of code, we'd need something to check on the node to identify it as a value item for filtering, and we'd also need a way for them to indicate whether they support searching and multiple selection. Thoughts?